### ISSUE TITLE
Resumable session management

### ISSUE THREAD

**Issue by samurainate** [att/rcloud#2454]
When working in RCloud, if notebooks become disconnected due to connection interruption, network change or laptop sleep any work in progress is effectively lost.  Upon reconnect, the notebook is reloaded and a new R session is started.  The previous R session is never reattached.

When working remotely in a login shell, I use screen to multiplex multiple sessions.  If the shell connection is detached for any reason, screen allows me to resume sessions.  Screen like features for RCloud would greatly improve its usability.  Session restart on connection interruption is the main barrier to usability in RCloud.

I would like detachable, resumable sessions in RCloud.  Sessions and work should persist at the server regardless of disconnect.  Server admins may set policy to terminate inactive sessions after a period of detachment.  The default behavior on disconnect should be reconnect, reauthenticate if neccesary, and then resume.  

Multiplexing behavior is a plus, from a fresh login I would like to be able to see my active sessions and resume any of them, possibly detaching another instance of the browser application in another tab or on another machine.

**Comment by gordonwoodhull** [comment]
Yes, it's definitely something we want to do. Fairly difficult design problem with a lot of ramifications, especially when the notebook has installed and run JavaScript.

Previous discussion: #1803, #777.
